bofire/benchmarks/LookUpTableBenchmark.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+import pandas as pd
+
+from bofire.benchmarks.benchmark import Benchmark
+from bofire.data_models.domain.api import Domain
+
+
+class LookUpTableBenchmark(Benchmark):
+    """This class is a benchmark for discrete data where you can lookup a table for new inputs.
+    It takes in a file with inputs and outputs and returns output for matching inputs from that file.
+
+    Args:
+        Benchmark: Subclass of the Benchmark function class.
+    """
+
+    def __init__(
+        self,
+        domain: Domain,
+        LookUpTable: pd.DataFrame,
+        **kwargs,
+    ) -> None:
+        """Initializes Molecule_benchmark. A class that connects to categorical molecular input.
+
+        Args:
+            domain (Domain): Domain of the inputs and outputs
+            LookUpTable (pd.DataFrame): DataFrame containing the LookUp table.
+        """
+        super().__init__(**kwargs)
+
+        self._domain = domain
+        self.LookUpTable = LookUpTable
+        self.domain.validate_experiments(self.LookUpTable)
+
+    def _f(self, X: pd.DataFrame, **kwargs) -> pd.DataFrame:
+        """return output values for matching SMILE candidates.
+
+        Args:
+            X (pd.DataFrame): Input values with input columns only.
+
+        Returns:
+            pd.DataFrame: output values from the LookUpTable. Columns are ouput keys and valid_output keys.
+        """
+        location = []
+        for i in X.index:
+            condition = np.ones(len(self.LookUpTable), dtype=bool)
+            for k in self.domain.inputs.get_keys():
```

Comment:
This looping over the single entries is kind of cumbersome. Try something like this:

```
import pandas as pd

# create a sample dataframe
df = pd.DataFrame({'A': [1, 2, 3], 'B': ['a', 'b', 'c'], 'C': [True, False, True]})

# create a sample series
s = pd.Series([2, 'b', False])

# check if the values of the series show up in a row of the dataframe
mask = df.isin(s)

# get the index of the rows that match the values in the series
index = mask.index[mask.all(axis=1)]

print(index)
```

If it matches all values, you can then use the index to return via loc the correct output values.

Reply:
isin() does not work with input keys having similar categories. I used pd.merge which is also a nice substitute for for loops